### PR TITLE
Go: Add additional architectures

### PR DESCRIPTION
**Pull request by vcabbage**
Adds all architectures Go currently supports on Linux.

I'm not sure the way I did the groups is ideal, happy to change it however would be best.

Uses 'go build' instead of 'go tool compile' to more easily support additional architectures.

Thanks for taking a look!

### CODE REVIEW

File: etc/config/go.amazon.properties [hunk 1]
```diff
@@ -16,53 +16,194 @@ compiler.gccgo720.semver=7.2.0
...
 
-group.gl.compilers=6g141:gl172:gl185:gl187:gl192:gl194:gl1100:gl1101:gl1110:gl1120:gl1130:gl1140:gltip
+group.gl.compilers=6g141:gl172:gl185:gl187:gl192:gl194:gl1100:gl1101:gl1110:gl1120:gl1130:gl1140:386_gl114:gltip:386_gltip
 group.gl.versionFlag=version
 group.gl.compilerType=golang
 group.gl.supportsBinary=false
 group.gl.compilerName=GoLang
 group.gl.isSemVer=true
-group.gl.baseName=x86 gc
+group.gl.groupName=x86 GC
```
[the remark this answers is not in the document]
Comment (vcabbage) [reply etc/config/go.amazon.properties:25]: I thought I tried that, but there's a good chance I had made changes in the wrong properties file. I try again and update the PR.

**Comment by vcabbage** [comment]
@RabsRincon Thanks for reviewing. I don't think a PR in the infra repo is necessary. Go's standard toolchain can cross compile to any supported OS/architecture as long there's no C involved.

File: etc/config/go.amazon.properties (hunk 1 above)
[the remark this answers is not in the document]
Comment (vcabbage) [reply etc/config/go.amazon.properties:26]: Done

**Comment by RabsRincon** [comment]
I've tested the PR locally and it looks great. We'll have to make sure it works on beta, but I'll go ahead and merge it now. Thank you so much for taking the time to make this PR :)

**Comment by vcabbage** [comment]
Sounds good, thank you!

**Pull request merged**